### PR TITLE
add metadata button is back

### PR DESCRIPTION
**Pull request by tcnichol**
fixes the missing metadata button.

### CODE REVIEW

**Review by ddey2 — APPROVED** [review]
looks good

**Comment by tcnichol** [comment]
The metadata button was missing since setEnableAddMetadata should not have been added to the conditional for the button. It starts false and will be set to true by the button.

File: frontend/src/components/files/File.tsx [hunk 1]
```diff
@@ -275,7 +275,7 @@ export const File = (): JSX.Element => {
 	} else if (showNotFoundPage) {
 		return <PageNotFound />;
 	}
-
+	console.log("file role is", fileRole);
```
Comment (longshuicy) [inline frontend/src/components/files/File.tsx:278]: Could you remove that logg?

**Comment by longshuicy** [comment]
After I created some user metadata and then click "add metadata" it leads me to a blank page
image
image

**Comment by tcnichol** [comment]
@longshuicy 

I have not been able to replicate the blank screen after add metadata.

I tried to replicate the same situation - metadata type abstract and metadata text field, but it worked for me and the new metadata was added to the dataset.

**Comment by tcnichol** [comment]
I am able to replicate the blank screen error. Working to fix.

**Comment by tcnichol** [comment]
I think I fixed the empty screen problem after clicking add metadata.

datasetRole !== undefined works
datasetRole.role !== undefined does not, since you can't get .role from undefined.

File: frontend/src/components/metadata/widgets/MetadataTextField.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const MetadataTextField = (props) => {
 				/>
 			</Grid>
 			<Grid item xs={1} sm={1} md={1} lg={1} xl={1}>
-				{datasetRole.role !== undefined && datasetRole.role !== "viewer" ?
+				{datasetRole !== undefined && datasetRole.role !== "viewer" ?
```
Comment (longshuicy) [inline frontend/src/components/metadata/widgets/MetadataTextField.tsx:42]: you might want to keep the original logic?
`datasetRole !== undefined && datasetRole.role !== undefined && datasetRole.role !== "viewer`

Comment (ddey2) [reply frontend/src/components/metadata/widgets/MetadataTextField.tsx:42]: We definitely need the whole check as Chen mentioned. else it still shows blank page.

**Review by ddey2 — CHANGES_REQUESTED** [review]

File: frontend/src/components/metadata/widgets/MetadataTextField.tsx (hunk 2 above)
Comment (tcnichol) [reply frontend/src/components/metadata/widgets/MetadataTextField.tsx:42]: It is back in that order.

**Review by longshuicy — APPROVED** [review]

**Pull request merged**